doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md

```diff
@@ -199,53 +222,123 @@ The DHCP_SERVER_IPV4_PORT table includes the following:
 
+``` text
+# bridge-midplane.netdev
+[NetDev]
+Name=bridge-midplane
+Kind=bridge
 ```
-auto bridge_midplane
-iface bridge_midplane inet static
-    bridge_ports Dpu0 Dpu1
-    address 169.254.200.254
-    netmask 255.255.255.0
-    network 169.254.200.0
-    broadcast 169.254.200.255
+
+``` text
+# bridge-midplane.network
+[Match]
+Name=bridge-midplane
+
+[Network]
+Address=169.254.200.254/24
 ```
 
-### IP assignment flow ###
+``` text
+# midplane-network.network
+[Match]
+Name=dpu0 dpu1
 
-![image](./smart-switch-ip-assignment-ip-assignment-flow.png)
+[Network]
+Bridge=bridge-midplane
+```
 
-- (1) sonic-cfggen based on hwsku.json renders midplane bridge and DHCP server configuration.
-- (2) sonic-cfggen pushes configuration into the Config DB.
-- (3-4) DHCP server and DHCP relay containers upon start consume configuration from the config DB and start listening for a requests from DHCP clients.
-- (5) DHCP client on the DPU sends a request over the eth0 interfaces. The request comes to the DHCP relay through the midplane bridge. DHCP relay inserts option 82 into the request with the information about the interface from which the request came. DHCP relay forwards the packet to the DHCP server.
-- (6) The DHCP server sends a reply with the IP configuration to the DHCP client on the DPU.
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
+
+...
+```
+
+2. systemd-networkd helps to create the bridge-midplane interface and assign the specific IP address according to above configuration. Meanwhile, systemd-networkd will monitor the DPU PCIe interface. Once the PCIe interface is created, it will automatically add it into the bridge-midplane.
+3. midplane-network.service will be used to wait for the midplane bridge configured.
 
-### DPU ###
+``` text
+# midplane-network.service
 
-In the Smart Switch DPU, the PCIe interface should be used as a default management interface for communication with the switch and the outside world.
+[Unit]
+Description=Midplane network service
+Requires=systemd-networkd.service
+After=systemd-networkd.service
 
-It is the vendors' responsibility to make sure that the PCIe interface that is connected to the switch has given the `eth0` name.
+[Service]
+Type=oneshot
+User=root
+ExecStart=/usr/lib/systemd/systemd-networkd-wait-online -i bridge-midplane
 
-The IP address assignment for the default `eth0` management interface in the existing SONiC implementation is the following:
+[Install]
+WantedBy=multi-user.target
+```
+
+``` text
+# database.service
+
+[Unit]
+...
+Requires=midplane-network.service
+After=midplane-network.service
+
+...
+```
+
+#### DPU ####
+
+In the DPU side, the steps are almost similar to NPU. But the generated configuration is different but simpler.
+
+``` text
+# midplane-network.network
+[Match]
+Name=eth0-midplane
+
+[Network]
+DHCP=yes
```

Comment:
@Pterosaur are we using SONiC specific DHCP client?

Reply:
According to my experiment and a link: https://askubuntu.com/a/1047654/1708404 .I feel the systemd-networkd will use an in-built DHCP client by default.
Do you have any concerns about this behavior? Why we need to use the SONiC specific DHCP client for midplane network?